File: CS/HowToExdendApplicationModelAndSchema.Module.Win/WinModule.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;

using DevExpress.ExpressApp;

namespace HowToExdendApplicationModelAndSchema.Module.Win
{
    [ToolboxItemFilter("Xaf.Platform.Win")]
    public sealed partial class HowToExdendApplicationModelAndSchemaWindowsFormsModule : ModuleBase
    {
        public HowToExdendApplicationModelAndSchemaWindowsFormsModule()
        {
            InitializeComponent();
        }
        public override Schema GetSchema()
        {
            const string CommonTypeInfos =
            @"<Element Name=""Application"">
                <Element Name=""Views"" >
                    <Element Name=""ListView"" >
                    <Attribute Name=""IsGroupFooterVisible"" Choice=""True,False""/>
                        <Element Name=""Columns"" >
                            <Element Name=""ColumnInfo"" >
                            <Attribute Name=""GroupFooterSummaryType"" Choice=""Sum,Min,Max,Count,Average,Custom,None"" />
                            </Element>
                        </Element>
                    </Element>
                </Element>
            </Element>";
            return new Schema(new DictionaryXmlReader().ReadFromString(CommonTypeInfos));
        }
    }
}

File: VB/HowToExdendApplicationModelAndSchema.Module.Win/WinModule.vb
Imports System
Imports System.Text
Imports System.Collections.Generic
Imports System.ComponentModel

Imports DevExpress.ExpressApp

<ToolboxItemFilter("Xaf.Platform.Win")> _
Partial Public NotInheritable Class [HowToExdendApplicationModelAndSchemaWindowsFormsModule]
	Inherits ModuleBase
	Public Sub New()
		InitializeComponent()
    End Sub
    Public Overrides Function GetSchema() As Schema
        Const CommonTypeInfos As String = _
           "<Element Name=""Application"">" + _
              "<Element Name=""Views"" >" + _
                 "<Element Name=""ListView"" >" + _
                    "<Attribute Name=""IsGroupFooterVisible"" Choice=""True,False""/>" + _
                      "<Element Name=""Columns"" >" + _
                         "<Element Name=""ColumnInfo"" >" + _
                            "<Attribute Name=""GroupFooterSummaryType"" Choice=""Sum,Min,Max,Count,Average,Custom,None"" />" + _
                         "</Element>" + _
                      "</Element>" + _
                   "</Element>" + _
              "</Element>" + _
           "</Element>"
        Return New Schema(New DictionaryXmlReader().ReadFromString(CommonTypeInfos))
    End Function
End Class

